File: PR1/PR1_Ej3/HamburguesaSinGlutenBuilder.py
from HamburguesaBuilder import HamburguesaBuilder
import time
class HamburguesaSinGlutenBuilder(HamburguesaBuilder):
    def aniadePan(self):
        self.hamburguesa.pan = "Pan sin gluten"
        time.sleep(0.2)

    def aniadeLechuga(self):
        self.hamburguesa.lechuga = "Lechuga verde"
        time.sleep(0.1)

    def aniadeTomate(self):
        self.hamburguesa.tomate = "Tomate pera"
        time.sleep(0.1)

    def aniadeQuesoCabra(self):
        self.hamburguesa.quesoCabra = "Queso de cabra (sin trazas)"
        time.sleep(0.1)

    def aniadeCebolla(self):
        self.hamburguesa.cebolla = "Cebolla llorosa"
        time.sleep(0.1)

    def aniadePepinillos(self):
        self.hamburguesa.pepinillos = "Pepinillos"
        time.sleep(0.1)

    def aniadeBacon(self):
        self.hamburguesa.bacon = "Bacon grasiento"
        time.sleep(0.5)

    def aniadeCarne(self):
        self.hamburguesa.carne = "Carne Wagyu (sin trazas)"
        time.sleep(0.8)

File: PR2/pantalla_pedidos_hamburgueseria/lib/model/HamburguesaSinGlutenBuilder.dart
import 'HamburguesaBuilder.dart';
import 'Hamburguesa.dart';

class HamburguesaSinGlutenBuilder extends HamburguesaBuilder {
  @override
  Hamburguesa hamburguesa = Hamburguesa();

  @override
  void aniadePan() {
    hamburguesa.pan = "Pan sin gluten";
    sleepLong();
  }

  @override
  void aniadeLechuga() {
    hamburguesa.lechuga = "Lechuga verde";
    sleep();
  }

  @override
  void aniadeTomate() {
    hamburguesa.tomate = "Tomate pera";
    sleep();
  }

  @override
  void aniadeQuesoCabra() {
    hamburguesa.quesoCabra = "Queso de cabra (sin trazas)";
    sleep();
  }

  @override
  void aniadeCebolla() {
    hamburguesa.cebolla = "Cebolla llorosa";
    sleep();
  }

  @override
  void aniadePepinillos() {
    hamburguesa.pepinillos = "Pepinillos";
    sleep();
  }

  @override
  void aniadeBacon() {
    hamburguesa.bacon = "Bacon grasiento";
    sleepLong();
  }

  @override
  void aniadeCarne() {
    hamburguesa.carne = "Carne Wagyu (sin trazas)";
    sleepLong();
  }

  @override
  void aniadePrecio() {
    hamburguesa.precio = 6;
  }
}